api/src/main/java/org/openmrs/module/mksreports/reports/OutpatientConsultationReportManager.java

```diff
@@ -156,18 +151,22 @@ public ReportDefinition constructReportDefinition() {
 		parameterMappings.put("locationList", "${locationList}");
 		
 		List<Concept> questionConcepts = inizService.getConceptsFromKey("report.opdconsult.diagnosisQuestion.concepts");
-		for (Concept question : questionConcepts) {
-			for (Concept member : allDiags.getSetMembers()) {
+		for (Concept member : allDiags.getSetMembers()) {
+			List<CodedObsCohortDefinition> codedObsList = new ArrayList<>();
+			for (Concept question: questionConcepts) {
 				CodedObsCohortDefinition diag = new CodedObsCohortDefinition();
 				diag.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
 				diag.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
 				diag.addParameter(new Parameter("locationList", "Visit Location", Location.class, List.class, null));
 				diag.setOperator(SetComparator.IN);
+
 				diag.setQuestion(question);
-				
 				diag.setValueList(Arrays.asList(member));
-				opdConsult.addRow(member.getDisplayString(), diag, parameterMappings);
+				codedObsList.add(diag);
 			}
+			CodedObsCohortDefinition[] codedObs = new CodedObsCohortDefinition[codedObsList.size()];
+			codedObs = codedObsList.toArray(codedObs);
+			opdConsult.addRow(member.getDisplayString(), createCohortComposition(codedObs), parameterMappings);
```

Comment:
Can you not do?
```java
createCohortComposition(codedObsList)
```
Or if you need an array while you're with a `List`, let's just transform this in one instruction.

Reply:
I didn't know you can pass a List where multiple variable number of arguments is expected. But I didn't look at the signature either, I quickly assumed it was something like `CohortDefinition... definitions`. It is instead `Object... elements` so the list should have been accepted too.

That been said, I will have to look at how the CompositionCohortDefinition.initializeFromElements actually extracts the data. Because if it's expecting an array and we're sending in a List, we may have a runtime error.

Reply:
Yes I think you're right, it will not work, but I think a one liner is fine in that case. Such as:
```java
opdConsult.addRow(member.getDisplayString(), createCohortComposition( codedObsList.toArray(new CodedObsCohortDefinition[codedObsList.size()]) ), parameterMappings);
```